pandas/core/strings.py

```diff
@@ -303,8 +310,9 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
     ----------
     pat : string
         Character sequence or regular expression
-    repl : string
-        Replacement sequence
+    repl : string or callable
+        Replacement string or a callable, it's passed the match object and 
+        must return a replacement string to be used. See :func:`re.sub`.
     n : int, default -1 (all)
         Number of replacements to make from start
```

Comment:
can you add some examples to the doc-string?

also would be nice to have an example in text.rst